### ISSUE TITLE
Faster R-CNN: Towards Real-Time Object Detection with Region Proposal Networks (2015)

### ISSUE THREAD

**Issue by Yphy** [Yphy/Paper-Review#2]
## Introduction

[Faster R-CNN: Towards Real-Time Object Detection with Region Proposal Networks](https://arxiv.org/abs/1506.01497)

Fast R-CNN 이후 속도가 대폭 개선된 Faster R-CNN 논문이다.
Fast R-CNN의 개념을 잠깐 짚고 넘어가자. 기존 R-CNN은 2000여개의 RP들로 모두 convolution 연산을 하며 SVM, Regression 도 각각 수행해 시간소요가 컸다. 

### Fast R-CNN
![image](https://user-images.githubusercontent.com/47969237/112596759-94816280-8e4f-11eb-88bf-ba4f2006748c.png)

* Fast R-CNN에서는 대신, Input Image에 한 번만 convolution 연산된 feature map에 **RoI Pooling**을 통해 얻은 feature vector가 fully connected layer을 거쳐 **multi-task loss** 사용,  classification(softmax), regression을 하도록 하여, 기존 Region proposal에서 CNN으로 들어가는 시간을 대폭 줄였다. 

* 하지만, selective search를 수행하는 region proposal 부분이 외부에 존재하기 때문에 inference에서 bottleneck을 일으켜 이 부분의 시간은 줄이지 못했다. 또한 CPU기반으로 실행되기 때문에 여전히 느리다.

# Faster R-CNN
> [Abstract] we introduce a **Region Proposal Network (RPN)** that shares full-image convolutional features with the detection network, thus enabling nearly cost-free region proposals

Faster R-CNN은 RPN을 도입하여 이 문제를 해결한다.
* *Faster R-CNN is a single, unified network for object detection. The RPN module serves as the ‘attention’ of this unified network.*  이 말은 공감이 안된다..

# Region Proposal Network
![333](https://user-images.githubusercontent.com/47969237/112596917-cd213c00-8e4f-11eb-88bc-60a40336dc88.JPG)

* RPN은 이미지를 인풋으로 받아 rectangular object proposals(rp) 를 반환하는 네트워크다.
이미지는 pretrained Imagnet으로 한 번 convolution 연산을 거쳐 피쳐맵을 반환한다. 여기까지는 Fast R-CNN과 같음.

* 이 패쳐맵 위에 한 번 더 convolution을 해주는데 nxn spatial window sliding 을 하면서(논문에서는 n=3),  window마다 고정된 anchor를 생성하며 region proposal을 구한다.

* 이후 생성된 feature map에 1x1 convolution 을 통해,  wxhx(nxnx2k),  wxhx(nxnx4k) 의 2개의 ouput을 생성한다(k는 anchor). 2k scores는 cls로 object가 속해있는지 여부 판단, 우측의 reg layer은 4k의 좌표다.
* 1x1 convolution을 통해 parameter 수를 줄이면서도 비선형성 증가는 덤..

![99](https://user-images.githubusercontent.com/47969237/112597159-1a9da900-8e50-11eb-994d-10422cf8dd0c.JPG)

* anchor마다 IoU를 구해서 가장 높거나 0.7이상인 anchor는 positive, 반대로 0.3이하는 negative labeling을 하고, 중간값들은 training에 이용되지 않는다.
